compiler/noirc_frontend/src/parser/parser/modifiers.rs

```diff
@@ -0,0 +1,38 @@
+use noirc_errors::Span;
+
+use crate::{ast::ItemVisibility, token::Keyword};
+
+use super::Parser;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(crate) struct Modifiers {
+    pub(crate) visibility: ItemVisibility,
+    pub(crate) visibility_span: Span,
+    pub(crate) unconstrained: Option<Span>,
+    pub(crate) comptime: Option<Span>,
+    pub(crate) mutable: Option<Span>,
+}
+
+impl<'a> Parser<'a> {
+    /// Modifiers = 'unconstrained'? ItemVisibility 'comptime'? 'mut'?
+    pub(crate) fn parse_modifiers(&mut self, allow_mutable: bool) -> Modifiers {
+        let unconstrained = self.parse_modifier(Keyword::Unconstrained);
+
+        let start_span = self.current_token_span;
+        let visibility = self.parse_item_visibility();
+        let visibility_span = self.span_since(start_span);
+
+        let comptime = self.parse_modifier(Keyword::Comptime);
+        let mutable = if allow_mutable { self.parse_modifier(Keyword::Mut) } else { None };
+
+        Modifiers { visibility, visibility_span, unconstrained, comptime, mutable }
+    }
+
+    fn parse_modifier(&mut self, keyword: Keyword) -> Option<Span> {
+        if self.eat_keyword(keyword) {
+            Some(self.previous_token_span)
+        } else {
+            None
+        }
+    }
```

Comment:
Seems like this should be `accept_modifier` or similar? Other `parse_*` methods may error if the thing is not present so we should be a bit more consistent at least for base cases like this.

Reply:
To solve this I inlined `parse_modifier` into callsites. The code ends up being 1 line longer, so that helper function wasn't brining much value.